cloudfront_utils.py

```diff
@@ -0,0 +1,58 @@
+import dns
+import ipaddress
+import json
+import sys
+
+import requests
+
+
+def get_cloudfront_ip_ranges():
+    cloudfront_ip_ranges_url = 'https://d7uri8nf7uskq.cloudfront.net/tools/list-cloudfront-ips'
+    cloudfront_backup_ip_ranges_url = 'https://ip-ranges.amazonaws.com/ip-ranges.json'
+    ip_ranges = []
+
+    try:
+        print('[*] Retrieving Cloudfront IP ranges from {}'.format(cloudfront_ip_ranges_url))
+        page_content = requests.get(cloudfront_ip_ranges_url, timeout=10)
+        ip_ranges_json = json.loads(page_content.text)
+        ip_ranges = sorted({x for v in ip_ranges_json.values() for x in v})
+
+    except requests.exceptions.RequestException:
+        sys.stderr.write('[-] Failed to retrieve Cloudfront IP ranges from cloudfront - trying alternate source')
+
+        try:
+            print('[*] Retrieving Cloudfront IP ranges from {}'.format(cloudfront_backup_ip_ranges_url))
+            page_content = requests.get(cloudfront_backup_ip_ranges_url, timeout=10)
+            ip_ranges_json = json.loads(page_content.text)
+            filtered_v4_ip_ranges = [x['ip_prefix'] for x in ip_ranges_json['prefixes'] if x['service'] == 'CLOUDFRONT']
+            filtered_v6_ip_ranges = [x['ipv6_prefix'] for x in ip_ranges_json['ipv6_prefixes'] if x['service'] == 'CLOUDFRONT']
+            ip_ranges = filtered_v4_ip_ranges + filtered_v6_ip_ranges
+        except requests.exceptions.RequestException:
+            sys.stderr.write('[-] Failed to retrieve Cloudfront IP ranges')
+            print('Exiting.')
+            exit(1)
+
+    return ip_ranges
+
+def is_cloudfront_ip(ip):
+    if getattr(is_cloudfront_ip, 'cloudfront_subnets', None) is None:
```

Comment:
This avoids pulling CloudFront IPs when the flag isn't used. It might be nice to modify CloudFlare support in a similar way. 

Right now these lists are generated on CLI run due to the global variable + import of cloudflare_utils. This occurs even if you just run the help command.

e.g 
```bash
python3 cloudflair.py --help                                       
[*] Retrieving Cloudflare IP ranges from https://www.cloudflare.com/ips-v4
```